src/main/java/dev/revere/alley/core/profile/listener/ProfileListener.java

```diff
@@ -112,9 +112,12 @@ private void onPlayerInteract(PlayerInteractEvent event) {
         ProfileService profileService = AlleyPlugin.getInstance().getService(ProfileService.class);
         Profile profile = profileService.getProfile(player.getUniqueId());
 
-        if (profile.getState() == ProfileState.LOBBY
-                || profile.getState() == ProfileState.EDITING
-                || profile.getState() == ProfileState.SPECTATING) {
+        ProfileState state = profile.getState();
+
+        if (state == ProfileState.LOBBY
+                || state == ProfileState.WAITING
+                || state == ProfileState.EDITING
+                || state == ProfileState.SPECTATING) {
```

Comment:
go through all of these and add tournament check when thats out (leave pr for now)